### ISSUE TITLE
LSP detects `go.mod` files as `lprolog`, not `gomod`

### ISSUE THREAD

**Issue by jpmcb** [neovim/neovim#17512]
### Neovim version (nvim -v)

0.6.1

### Vim (not Nvim) behaves the same?

n/a

### Operating system/version

Linux ubuntu-focal 5.4.0-77-generic

### Terminal name/version

alacritty 0.10.1-rc1

### $TERM environment variable

screen-256color

### Installation

appimage

### How to reproduce the issue

1. In a brand new VM, I've downloaded nvim as the appimage:

```
$ ./nvim.appimage --version
NVIM v0.6.1
Build type: RelWithDebInfo
...
```

2. Added the following `~/.config/nvim/init.lua` configuration (all copied from the `nvim-lspconfig` repo) in order to setup very basic LSP for go.:
```lua
local use = require('packer').use
require('packer').startup(function()
  use 'wbthomason/packer.nvim' -- Package manager
  use 'neovim/nvim-lspconfig' -- Collection of configurations for the built-in LSP client
end)

require'lspconfig'.gopls.setup{}
```
also ensured the `:PackerSync` was ran to get the necessary plugins.

3. Installed the gopls LSP:
```
$ go install golang.org/x/tools/gopls@latest
$ which gopls
/home/vagrant/go/bin/gopls
```

4. Cloned a basic Go project
```
$ git clone https://github.com/spf13/cobra
```

5. Open a go file
```
$ ./nvim.appimage
```
and check LSP info `:LspInfo`
```
Langauge client log: /hom/vagrant/.cache/nvim/lsp.log
Detected filetype: go

1 client(s) attached to this buffer:

Client: gopls (id: 1, pid: 21311, bufnr: [1])
  autostart:
  rootdirectory: /hom/vagrant/cobra
  cmd: gopls

Configuraed servers list: gopls
```

6. Exit. Open the `go.mod` file in cobra:
```
$ ./nvim.appimage
```
and check LSP info again: `:LspInfo`
```
Langauge client log: /hom/vagrant/.cache/nvim/lsp.log
Detected filetype: lprolog

0 client(s) attached to this buffer:

Configured servers list: gopls
```

### Expected behavior

I would have expected to see `go` or `gomod` when attaching to a `go.mod` file, not `lprolog`.

On doing a quick search of the `neovim/neovim` codebase, I do see this in the `lprolog.vim` file:

https://github.com/neovim/neovim/blob/15004473b531e37a6ff3aeca9e3bfeaaa1487d9e/runtime/syntax/lprolog.vim#L3

Is there some kind of clash happening when detecting `go.mod` files?

### Actual behavior

When a buffer that's opened a `go.mod` file opens, the configured `gopls` LSP does not attach. This means that with more plugins, there are no completions or other behavior that takes advantage of the `gopls` LSP for `go.mod`

**Comment by zeertzjq** [comment]
Already fixed by #16757

**Comment by jpmcb** [comment]
Thanks much - updated to [neovim nightly](https://github.com/neovim/neovim/releases/nightly) and now I see `Detected filetype: gomod` in LspInfo 👏🏼
